driver/src/price_estimation/kraken/api.rs

```diff
@@ -0,0 +1,272 @@
+            .into_result()
+            .unwrap()
+    }
+
+    #[test]
+    fn parse_assets_json() {
+        // Sample retrieved from https://api.kraken.com/0/public/Assets?asset=DAI,ETH
+        let value: HashMap<String, Asset> = deserialize(
+            r#"{"error":[],"result":{"DAI":{"aclass":"currency","altname":"DAI","decimals":10,"display_decimals":5},"XETH":{"aclass":"currency","altname":"ETH","decimals":10,"display_decimals":5}}}"#,
+        );
+        assert_eq!(
+            value,
+            hash_map! {
+                "DAI" => Asset::new("DAI"),
+                "XETH" => Asset::new("ETH"),
+            }
+        );
+    }
+
+    #[test]
+    fn parse_asset_pairs_json() {
+        // Sample retrieved from https://api.kraken.com/0/public/AssetPairs?pair=DAIUSD,ETHUSD
+        let value: HashMap<String, AssetPair> = deserialize(
+            r#"{"error":[],"result":{"DAIUSD":{"altname":"DAIUSD","wsname":"DAI\/USD","aclass_base":"currency","base":"DAI","aclass_quote":"currency","quote":"ZUSD","lot":"unit","pair_decimals":5,"lot_decimals":8,"lot_multiplier":1,"leverage_buy":[],"leverage_sell":[],"fees":[[0,0.2],[50000,0.16],[100000,0.12],[250000,0.08],[500000,0.04],[1000000,0]],"fees_maker":[[0,0.2],[50000,0.16],[100000,0.12],[250000,0.08],[500000,0.04],[1000000,0]],"fee_volume_currency":"ZUSD","margin_call":80,"margin_stop":40},"XETHZUSD":{"altname":"ETHUSD","wsname":"ETH\/USD","aclass_base":"currency","base":"XETH","aclass_quote":"currency","quote":"ZUSD","lot":"unit","pair_decimals":2,"lot_decimals":8,"lot_multiplier":1,"leverage_buy":[2,3,4,5],"leverage_sell":[2,3,4,5],"fees":[[0,0.26],[50000,0.24],[100000,0.22],[250000,0.2],[500000,0.18],[1000000,0.16],[2500000,0.14],[5000000,0.12],[10000000,0.1]],"fees_maker":[[0,0.16],[50000,0.14],[100000,0.12],[250000,0.1],[500000,0.08],[1000000,0.06],[2500000,0.04],[5000000,0.02],[10000000,0]],"fee_volume_currency":"ZUSD","margin_call":80,"margin_stop":40}}}"#,
+        );
+        assert_eq!(
+            value,
+            hash_map! {
+                "DAIUSD" => AssetPair::new("DAI", "ZUSD"),
+                "XETHZUSD" => AssetPair::new("XETH", "ZUSD"),
+            }
+        );
+    }
+
+    #[test]
+    fn parse_ticker_infos_json() {
+        // Sample retrieved from https://api.kraken.com/0/public/Ticker?pair=DAIUSD,ETHUSD
+        let value: HashMap<String, TickerInfo> = deserialize(
+            r#"{"error":[],"result":{"DAIUSD":{"a":["1.00105000","3091","3091.000"],"b":["0.99989000","628","628.000"],"c":["0.99986000","115.05510869"],"v":["267638.81097379","587455.25359912"],"p":["0.99938164","0.99862453"],"t":[1219,2709],"l":["0.99696000","0.99608000"],"h":["1.00300000","1.00300000"],"o":"0.99828000"},"XETHZUSD":{"a":["257.73000","124","124.000"],"b":["257.72000","1","1.000"],"c":["257.81000","6.60000000"],"v":["99246.07873952","130832.28052933"],"p":["251.77023","248.47638"],"t":[13177,18557],"l":["238.06000","230.57000"],"h":["259.48000","259.48000"],"o":"238.06000"}}}"#,
```

Comment:
Seems like the only relevant values here are `"p":["0.99938164","0.99862453"]`. What is all this other stuff?

Reply:
As mentioned in the comment: `Sample retrieved from https://api.kraken.com/0/public/Ticker?pair=DAIUSD,ETHUSD`. Its a sample JSON document returned by the Kraken API. This test just makes sure that it can be correctly parsed.

Since our implementation only cares about the `"p":["0.99938164","0.99862453"]`, that's all we parse from the JSON document.

Reply:
But what is the "p" does it stand for "price"? "pair"? why does it contain exactly two values?

Reply:
The `p` comes from the Kraken API: 
> ```
> <pair_name> = pair name
>     a = ask array(<price>, <whole lot volume>, <lot volume>),
>     b = bid array(<price>, <whole lot volume>, <lot volume>),
>     c = last trade closed array(<price>, <lot volume>),
>     v = volume array(<today>, <last 24 hours>),
>     p = volume weighted average price array(<today>, <last 24 hours>),
>     t = number of trades array(<today>, <last 24 hours>),
>     l = low array(<today>, <last 24 hours>),
>     h = high array(<today>, <last 24 hours>),
>     o = today's opening price
> ```

Reply:
This is very nice thanks for going the extra mile!

Reply:
Whoever wrote that API must be a python dev without any guidance (cf. single character identifiers).